src/features/user/TreeMapper/Analytics/components/Map/index.tsx

```diff
@@ -299,10 +300,19 @@ export const MapContainer = () => {
           return;
         }
       }
-      fetchProjectLocations();
     }
   }, [project, species, queryType, fromDate, toDate]);
 
+  useDebouncedEffect(
+    () => {
+      if (project && species) {
+        fetchProjectLocations();
+      }
+    },
+    500,
+    [project, species, queryType, fromDate, toDate]
+  );
```

Comment:
Wouldn't the combination of these 2 useEffects (295-302) and (306-314) be unpredictable? A `searchDateError` will not prevent the debounced effect from running.

Reply:
Resolved in #2062